src/main/java/org/terasology/Lakes/LakeProvider.java

```diff
@@ -160,17 +159,17 @@ private void generateUndergroundLake(Vector3i origin, LakeFacet facet, DensityFa
                 - square((pos.y - origin.y) / depth)
                 - square((pos.z - origin.z) / width);
             if (lakeness > 0) {
-                if (!densityFacet.getWorldRegion().encompasses(pos) || densityFacet.getWorld(pos) <= 0) {
+                if (!densityFacet.getWorldRegion().contains(pos) || densityFacet.getWorld(pos) <= 0) {
                     // The lake breaches the surface. Abort.
                     return;
                 }
                 content.add(pos);
-                frontier.add(Vector3i.north().add(pos));
-                frontier.add(Vector3i.south().add(pos));
-                frontier.add(Vector3i.east().add(pos));
-                frontier.add(Vector3i.west().add(pos));
-                frontier.add(Vector3i.up().add(pos));
-                frontier.add(Vector3i.down().add(pos));
+                frontier.add(new Vector3i(0,0,1).add(pos));
```

Comment:
This would be a replacement... 🤔 
```suggestion
                frontier.add(Direction.FORWARD.asVector3i().add(pos, new Vector3i()));
```